pkg/datapath/linux/node.go

```diff
@@ -1544,31 +1533,80 @@ func (n *linuxNodeHandler) NodeCleanNeighbors() {
 		return
 	}
 
-	var successRemoval, errRemoval int
+	var neighSucceeded, neighErrored int
+	var which string
 	for _, neigh := range neighList {
-		err := netlink.NeighDel(&neigh)
+		var err error
+		// If this is a non-static neighbor entry, it will be GC'ed by
+		// the kernel eventually. Older Cilium versions might have left-
+		// overs installed as NUD_PERMANENT.
+		if (neigh.State&netlink.NUD_PERMANENT) == 0 &&
+			(neigh.Flags&netlink.NTF_EXT_LEARNED) == 0 {
```

Comment:
Nit: I don't think the parenthesis are required here in Go:
```suggestion
		if neigh.State&netlink.NUD_PERMANENT == 0 &&
			neigh.Flags&netlink.NTF_EXT_LEARNED == 0 {
```